### PR TITLE
Docs: Game Server Allocation Details 

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup

/kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by Kalaiselvi84** [comment]
@markmandel, I have updated the `site/content/en/docs/Reference/gameserverallocation.md` file with information about game server allocation. Could you please review it and share your thoughts?

File: site/content/en/docs/Reference/gameserverallocation.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ description: "A `GameServerAllocation` is used to atomically allocate a GameServ
 weight: 30
 ---
 
+{{% feature publishVersion="1.38.0" %}}
```
Comment (markmandel) [inline site/content/en/docs/Reference/gameserverallocation.md:10]: No need for a feature flag - feature flags are only for when we talk about a feature that hasn't been released yet. Updated/improvements to existing docs for existing features don't need it.

Comment (markmandel) [inline site/content/en/docs/Reference/gameserverallocation.md:10]: I like this place better for this documentation improvement! 👍🏻

File: site/content/en/docs/Reference/gameserverallocation.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ description: "A `GameServerAllocation` is used to atomically allocate a GameServ
...
+When we have a fleet of game servers ready, allocation is the process of selecting a one of these servers for use. This is achieved using a `GameServerAllocation`. The allocation process marks the chosen game server for gameplay and marks it as `Allocated`. This ensures that the server is reserved for active players and not considered `ready` for other allocations until `SDK.Shutdown()` is called, or it is manually deleted.
```
Comment (markmandel) [inline site/content/en/docs/Reference/gameserverallocation.md:11]:
```suggestion
Allocation is the process of selecting the optimal `GameServer` that matches the filters defined in the `GameServerAllocation` specification below, and returning its details.

A successful Alloction moves the `GameServer` to the `Allocated` state, which demarcates that it is currently active, likely with players on it, and should not be removed until SDK.Shutdown() is called, or it is explicitly manually deleted.
```

Some tweaks to the language.

@roberthbailey, @Kalaiselvi84  WDYT?

Comment (Kalaiselvi84) [reply site/content/en/docs/Reference/gameserverallocation.md]: I've updated this change. Shall we include this info in FAQ page?

Comment (Kalaiselvi84) [reply site/content/en/docs/Reference/gameserverallocation.md]: So the answer is `no`.  Shall we merge this PR?

**Review by markmandel — APPROVED** [review]

**Pull request merged**